### PR TITLE
Add support for system metrics

### PR DESCRIPTION
**Pull request by jgiannuzzi**
This PR enables support for system metrics by reverting some of the code removals we did in #31 (based on #3), adapting them for MLFlow keys (starting with `system/` instead of `__system__` and covering different aspects), and fixing the key ordering (which was broken in the UI and relied on the server originally).

Fixes https://github.com/G-Research/fasttrackml/issues/536.

### CODE REVIEW

File: src/src/config/systemMetrics/systemMetrics.ts [hunk 1]
```diff
@@ -1,12 +1,60 @@
...
-  __system__disk_percent: 'Disk (%)',
-  __system__gpu: 'GPU (%)',
-  __system__gpu_memory_percent: 'GPU Memory (%)',
-  __system__gpu_power_watts: 'GPU Power (W)',
-  __system__gpu_temp: 'GPU Temperature (°C)',
+  'system/cpu_utilization_percentage': 'CPU (%)',
+  'system/disk_available_megabytes': 'Disk Available (MB)',
+  'system/disk_usage_megabytes': 'Disk (MB)',
+  'system/disk_usage_percentage': 'Disk (%)',
+  'system/gpu_0_memory_usage_megabytes': 'GPU 0 Memory (MB)',
```
Comment (dave-gantenbein) [inline src/src/config/systemMetrics/systemMetrics.ts:8]: Couldn't this be built more cleanly with an iterator?

File: src/src/services/models/runs/runDetailAppModel.ts [hunk 2]
```diff
@@ -146,7 +146,7 @@ function processRunBatchData(
       }),
       sortKey: `${run.name}${contextName}`,
     };
-    if (run.name.startsWith('__system__')) {
+    if (run.name.startsWith('system/')) {
```
Comment (dave-gantenbein) [inline src/src/services/models/runs/runDetailAppModel.ts:149]: This is the third time I'm seeing this magic string in this PR - could we extract a constant somehow?

File: src/src/types/utils/formatSystemMetricName.d.ts [hunk 3]
```diff
@@ -1,10 +1,58 @@
...
-  __system__gpu: string;
-  __system__gpu_memory_percent: string;
-  __system__gpu_power_watts: string;
-  __system__gpu_temp: string;
-};
+  'system/cpu_utilization_percentage': string;
+  'system/disk_available_megabytes': string;
+  'system/disk_usage_megabytes': string;
+  'system/disk_usage_percentage': string;
+  'system/gpu_0_memory_usage_megabytes': string;
```
Comment (dave-gantenbein) [inline src/src/types/utils/formatSystemMetricName.d.ts:6]: Same comment re: tidying this w/an iterator...

File: src/src/config/systemMetrics/systemMetrics.ts (hunk 1 above)
Comment (jgiannuzzi) [reply src/src/config/systemMetrics/systemMetrics.ts:8]: I am not an expert in typescript, so I don't know 😞
I wanted to make sure we can ship this!

File: src/src/services/models/runs/runDetailAppModel.ts (hunk 2 above)
Comment (jgiannuzzi) [reply src/src/services/models/runs/runDetailAppModel.ts:149]: This is based on the existing UI code from Aim at this stage. Extracting a constant would make total sense, but I wanted to keep this as close as possible to the current upstream logic.

File: src/src/types/utils/formatSystemMetricName.d.ts (hunk 3 above)
Comment (jgiannuzzi) [reply src/src/types/utils/formatSystemMetricName.d.ts:6]: same reply as above: if there is a way to do this cleanly in typescript, I'll take it!

**Review by suprjinx — APPROVED** [review]

**Pull request merged**